targets/mbedos5/tools/generate_pins.py

```diff
@@ -0,0 +1,224 @@
+            operators[ast.BitOr] = lambda a, b: a | b
+            operators[ast.LShift] = lambda a, b: a << b
+            operators[ast.RShift] = lambda a, b: a << b
+            evaluator = SimpleEval(DEFAULT_OPERATORS )
+
+            for pin in node.type.values.enumerators:
+                expr = c_gen.visit(pin.value)
+
+                if "(int)" in expr:
+                    expr = expr.replace('(int)', '')
+
+                if expr in pins:
+                    pins[pin.name] = pins[expr]
+                else:
+                    pins[pin.name] = evaluator.eval(expr.strip())
+
+            return pins
+
+def enumerate_pins(c_source_file, include_dirs, definitions):
+    """
+    Enumerate pins specified in PinNames.h, by looking for a PinName enum
+    typedef somewhere in the file.
+    """
+    definitions += ['__attribute(x)__=', '__extension__(x)=', 'register=', '__IO=', 'uint32_t=unsigned int']
+
+    gcc_args = ['-E', '-fmerge-all-constants'] 
+    gcc_args += ['-I' + directory for directory in include_dirs] 
+
+    gcc_args += ['-D' + definition for definition in definitions]
+    ast = parse_file(c_source_file,
+                    use_cpp=True,
+                    cpp_path='arm-none-eabi-gcc',
+                    cpp_args=gcc_args,
+                    parser=GnuCParser())
+
+    # now, walk the AST
+    v = TypeDeclVisitor(['PinName'])
+    return v.visit(ast)
+
+
+if __name__ == "__main__":
+    if not os.path.exists('./mbed-os'):
+        print("Fatal: mbed-os directory does not exist.")
+        print("Try running 'make getlibs'")
+        sys.exit(1)
+
+    description = """
+    Generate pins.js for a specified mbed board, using target definitions from the
+    mbed OS source tree.
+    """
+
+    parser = argparse.ArgumentParser(description=description)
+
+    parser.add_argument('board', help='mbed board name')
+    parser.add_argument('-o',
+                        help='Output JavaScript file (default: %(default)s)',
+                        default='js/pins.js',
+                        type=argparse.FileType('w'))
+    parser.add_argument('-c',
+                        help='Output C++ file (default: %(default)s)',
+                        default='source/pins.cpp',
+                        type=argparse.FileType('w'))
+
+    args = parser.parse_args()
+    board_name = args.board.upper()
+
+    target = Target(board_name)
+
+    directory_labels = ['TARGET_' + label for label in target.get_labels()] + target.macros
+
+    targets_dir = os.path.join('.', 'mbed-os', 'hal', 'targets')
+    hal_dir = os.path.join(targets_dir, 'hal')
+
+    pins_file = find_file(hal_dir, directory_labels, 'PinNames.h')
+
+
+    includes = enumerate_includes(targets_dir, directory_labels)
+    defines = list(directory_labels)
+
+    # enumerate pins from PinNames.h
+    pins = enumerate_pins(pins_file, ['./tools'] + list(includes), defines)
+
+    out_file = '\r\n'.join(['var %s = %s;' % pin for pin in pins.iteritems()])
```

Comment:
We should use `items` instead of `iteritems` as in python3 the iter one was removed. Just to be a bit future proof. This applies to the other `iter*` methods on lists/dicts.